Modify rule S6200: Updated rspec so it works for C and ObjC (#445)
SonarSource/rspec: rules/S6200/cfamily/rule.adoc

``++volatile++`` can be used to qualify many objects in C and {cpp}, but only a few of the possible places have a well-defined meaning (global variables and local variables for instance).

> There is no well-defined meaning to the use of volatile to qualify a function return type, a function parameter. 

Reviewer: ```suggestion
There is no well-defined meaning to the use of volatile to qualify a function return type or a function parameter. 
```

> Furthermore for the structured-binding the volatile qualifier appertains to the decomposed object, that cannot be referred.

Reviewer: ```suggestion
Furthermore, for structured bindings, the volatile qualifier appertains to the decomposed object which cannot be referred to.
```

Since {cpp}20, these uses are deprecated, but even before you should not use volatile in those places.

> This rule raises issue for a volatile qualified function return type, function parameter, and structured binding (available in {cpp} only since {cpp}17).

Reviewer: ```suggestion
This rule raises an issue for a volatile qualified function return type, function parameter, and structured binding (available in {cpp} since {cpp}17).
```
